src/runtime/HalideBuffer.h

```diff
@@ -413,6 +412,37 @@ class Buffer {
         }
     }
 
+    /** slice a single dimension without handling device allocation. */
+    void slice_host(int d, int pos) {
+        assert(d >= 0 && d <= dimensions());
```

Comment:
Is `d <= dimensions()` correct? Original code seems to have had a comment `pos <= dim(d).max()` which would indicate using `<`, not `<=`.

Reply:
Nope. Good catch.